skada/datasets/_base.py

```diff
@@ -314,10 +314,31 @@ def pack_lodo(self, return_X_y: bool = True) -> PackedDatasetType:
         )
 
     def __str__(self) -> str:
-        return f"DomainAwareDataset(domains={list(self.domain_names_.keys())})"
+        return f"DomainAwareDataset(domains={self.get_domain_representation()})"
 
     def __repr__(self) -> str:
-        return self.__str__()
+        head = self.__str__()
+        body = [f"Number of domains: {len(self.domains_)}"]
+        body.append(f"Total size: {sum(len(tup[0]) for tup in self.domains_)}")
+        output = "\n".join([head] + body)
+        return output
+    
+    def get_domain_representation(self, max_domains=5, max_length=50):
```

Comment:
I suggest renaming the function to `_get_domain_representation` to emphasize its role as an internal function. This change will help clarify that it should not be considered part of the class's public API.